### PR TITLE
1.8.4 rebase

### PR DESCRIPTION
**Pull request by liggitt**
Based on HEAD commit from https://github.com/openshift/kubernetes/pull/47

No interesting origin changes

One upstream API change (ScaleIO PV secret namespace split)

### CODE REVIEW

**Comment by liggitt** [comment]
cc @openshift/sig-master

File: pkg/cmd/server/origin/openshift_apiserver.go [hunk 1]
```diff
@@ -617,9 +616,8 @@ func apiLegacyV1(all map[string]rest.Storage) *genericapiserver.APIGroupInfo {
 		VersionedResourcesStorageMap: map[string]map[string]rest.Storage{},
 		Scheme: kapi.Scheme,
 		// version.ParameterCodec = runtime.NewParameterCodec(kapi.Scheme)
-		ParameterCodec:              kapi.ParameterCodec,
-		NegotiatedSerializer:        kapi.Codecs,
-		SubresourceGroupVersionKind: map[string]schema.GroupVersionKind{},
```
Comment (mfojtik) [inline pkg/cmd/server/origin/openshift_apiserver.go]: SubresourceGroupVersionKind for scale is not needed anymore?

Comment (mfojtik) [reply pkg/cmd/server/origin/openshift_apiserver.go]: seems like the test/cmd failing with error: no kind "Scale" is registered for version "extensions/v1beta1"

**Comment by mfojtik** [comment]
@liggitt working on the Scale thing... seems troublesome: https://gist.github.com/soltysh/d3074edec8c7d67c6010acb33e6a295f

I passed the 'get' error with this patch: https://github.com/mfojtik/origin/commit/c8826603bfbf12403ed089fd124765bb906ddd04

File: pkg/cmd/server/origin/openshift_apiserver.go (hunk 1 above)
Comment (mfojtik) [reply pkg/cmd/server/origin/openshift_apiserver.go]: see https://github.com/liggitt/origin/pull/14

Comment (liggitt) [reply pkg/cmd/server/origin/openshift_apiserver.go]: I see the errors too. I don't think manual registration is the answer... trying to track down what changed

Comment (soltysh) [reply pkg/cmd/server/origin/openshift_apiserver.go]: I've tested and verified this approach works: https://github.com/liggitt/origin/pull/15

**Comment by mfojtik** [comment]
@liggitt can you rebase this so we can land it?

**Comment by liggitt** [comment]
waiting for https://github.com/openshift/origin/pull/17520

**Comment by liggitt** [comment]
and then need to re-move upstreams to openshift/kubernetes/release-1.8.1, rebase openshift/kubernetes/release-1.8.4, then I'll update this

**Comment by deads2k** [comment]
lgtm.  Tag once you get the deps fanned out.

**Comment by liggitt** [comment]
1.8.5 was out. closing in favor of https://github.com/openshift/origin/pull/17849

**Pull request closed without merging**